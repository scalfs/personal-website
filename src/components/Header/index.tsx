import { Popover, Transition as TransitionHUI } from '@headlessui/react'
import Button from 'components/Button'
import Transition, { Transitions } from 'components/Transition'
import { IconMenu, IconX } from 'icons'
import { Fragment } from 'react'

const navigation = [
  { name: 'About', href: '#about' },
  { name: 'Experience', href: '#experience' },
  { name: 'Contact', href: '#contact' }
]

const Header = () => (
  <header className="relative mx-auto max-w-5xl">
    <Popover
      as="nav"
      className="absolute w-full px-4 sm:px-6 sm:text-lg md:px-8"
    >
      <Transition type="none">
        <div className="flex items-center pt-6 text-base">
          <div className="-mx-1 flex grow items-center justify-between">
            <Transition isChild type="fadedown">
              <a
                href="#"
                aria-label="Home"
                className="rounded-lg py-1 px-2 text-2xl font-semibold transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2"
              >
                ~/
              </a>
            </Transition>

            <Transition isChild type="fadedown" delay={500}>
              <Popover.Button
                title="Menu"
                className="h-10 w-10 rounded-md p-2 transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 sm:hidden"
              >
                <IconMenu />
              </Popover.Button>
            </Transition>
          </div>

          <ol className="hidden space-x-6 sm:flex">
            {navigation.map(({ name, href }, i) => (
              <li key={name}>
                <Transition isChild type="fadedown" delay={(i + 1) * 150}>
                  <a
                    href={href}
                    className="link-underline rounded-lg py-2 font-medium focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
                  >
                    {name}
                  </a>
                </Transition>
              </li>
            ))}
            <li>
              <Transition
                isChild
                type="fadedown"
                delay={(navigation.length + 1) * 150}
              >
                <Button href="/resume.pdf">Resume</Button>
              </Transition>
            </li>
          </ol>
        </div>
      </Transition>

      <TransitionHUI as={Fragment} {...Transitions.popover}>
        <Popover.Panel
          focus
          className="absolute inset-x-0 top-0 z-10 origin-top-right bg-slate-50 p-2 transition sm:hidden"
        >
          <div className="rounded-lg bg-slate-50 shadow-md ring-1 ring-black/5">
            <div className="mb-2 flex items-center justify-between px-5 pt-4">
              <Popover.Button
                aria-label="Home"
                className="-ml-3 rounded-lg py-1 px-3 text-2xl font-semibold transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 hover:text-slate-900 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2"
              >
                ~/
              </Popover.Button>

              <Popover.Button
                aria-label="Close Menu"
                className="-mr-2 h-10 w-10 rounded-lg p-2 transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 hover:text-slate-900 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2"
              >
                <IconX />
              </Popover.Button>
            </div>
            <ol className="mb-2 flex flex-col space-y-4 p-2">
              {navigation.map(({ name, href }) => (
                <li key={name}>
                  <a
                    href={href}
                    className="rounded-lg px-3 py-2 font-medium transition-colors duration-250 ease-in-out-cubic first-letter:text-lg hover:bg-blue-100 hover:text-slate-900 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2"
                  >
                    {name}
                  </a>
                </li>
              ))}
            </ol>
            <a
              href="/resume.pdf"
              className="block bg-slate-100 px-5 py-3 text-center text-lg font-medium text-blue-700 transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 focus:rounded-lg focus:outline-none focus:ring-2 focus:ring-inset focus:ring-blue-700"
            >
              Resume
            </a>
          </div>
        </Popover.Panel>
      </TransitionHUI>
    </Popover>
  </header>
)
export default Header
