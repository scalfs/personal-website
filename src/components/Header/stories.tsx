import { ComponentMeta, ComponentStory } from '@storybook/react'
import Header from '.'

export default {
  title: 'Header',
  component: Header
} as ComponentMeta<typeof Header>

const Story: ComponentStory<typeof Header> = () => <Header />

export const Default = Story.bind({})
