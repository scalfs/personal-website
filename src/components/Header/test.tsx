import { render, screen } from '@testing-library/react'

import Header from '.'

describe('<Header />', () => {
  it('should render all the nav options', async () => {
    const { container } = render(<Header />)

    expect(
      await screen.findByRole('button', { name: /Menu/i })
    ).toBeInTheDocument()
    expect(
      await screen.findByRole('link', { name: /Home/i })
    ).toBeInTheDocument()
    expect(
      await screen.findByRole('link', { name: /About/i })
    ).toBeInTheDocument()
    expect(
      await screen.findByRole('link', { name: /Experien/i })
    ).toBeInTheDocument()
    expect(
      await screen.findByRole('link', { name: /Contact/i })
    ).toBeInTheDocument()
    expect(
      await screen.findByRole('link', { name: /Resume/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
