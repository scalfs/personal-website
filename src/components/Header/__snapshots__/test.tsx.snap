// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render all the nav options 1`] = `
<header
  class="relative mx-auto max-w-5xl"
>
  <nav
    class="absolute w-full px-4 sm:px-6 sm:text-lg md:px-8"
  >
    <div
      class="flex items-center pt-6 text-base"
    >
      <div
        class="-mx-1 flex grow justify-between"
      >
        <div
          class="transition ease-in-out-cubic duration-300 opacity-0 -translate-y-4"
          style="transition-delay: undefinedms;"
        >
          <a
            aria-label="Home"
            class="rounded-md py-1 px-2 text-2xl font-semibold transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2"
            href="#"
          >
            ~/
          </a>
        </div>
        <div
          class="transition ease-in-out-cubic duration-300 opacity-0 -translate-y-4"
          style="transition-delay: 250ms;"
        >
          <button
            aria-expanded="false"
            class="h-10 w-10 rounded-md p-2 transition-colors duration-250 ease-in-out-cubic hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 sm:hidden"
            id="headlessui-popover-button-:r0:"
            title="Menu"
            type="button"
          >
            <svg
              fill="none"
              role="img"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Menu
              </title>
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
      </div>
      <ol
        class="hidden space-x-6 sm:flex"
      >
        <li>
          <div
            class="transition ease-in-out-cubic duration-300 opacity-0 -translate-y-4"
            style="transition-delay: 250ms;"
          >
            <a
              class="link-underline rounded-md py-2 font-medium focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
              href="#about"
            >
              About
            </a>
          </div>
        </li>
        <li>
          <div
            class="transition ease-in-out-cubic duration-300 opacity-0 -translate-y-4"
            style="transition-delay: 500ms;"
          >
            <a
              class="link-underline rounded-md py-2 font-medium focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
              href="#experience"
            >
              Experience
            </a>
          </div>
        </li>
        <li>
          <div
            class="transition ease-in-out-cubic duration-300 opacity-0 -translate-y-4"
            style="transition-delay: 750ms;"
          >
            <a
              class="link-underline rounded-md py-2 font-medium focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50"
              href="#contact"
            >
              Contact
            </a>
          </div>
        </li>
        <li>
          <div
            class="transition ease-in-out-cubic duration-300 opacity-0 -translate-y-4"
            style="transition-delay: 1000ms;"
          >
            <a
              class="rounded-lg border border-slate-300 px-5 py-3 font-semibold text-blue-700 transition-colors duration-250 ease-in-out-cubic hover:border-blue-700 hover:bg-blue-100 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:ring-offset-2 focus:ring-offset-slate-50 undefined"
              href="/resume.pdf"
              target="_blank"
            >
              Resume
            </a>
          </div>
        </li>
      </ol>
    </div>
  </nav>
</header>
`
